### ISSUE TITLE
Large amount of Direct Traffic investigation

### ISSUE THREAD

**Issue by stoweandi** [storj/storj#4605]
Since launching the proxy we've seen an abnormal amount of Direct Traffic (91% is way too much). We've done work to filter out bot traffic. Perhaps we need to filter out more...

AC:
- meet to discuss what could be the issues for the large amounts of bot traffic

> rule out what we've already know it's not
- Come up with next steps to investigate and list in this ticket

Query: https://metabase.datasci.storj.io/question/139-top-referral-sources-pie-chart

**Comment by wthorp** [comment]
@littleskunk can you comment on this?

**Comment by mobyvb** [comment]
@stoweandi the metabase query you shared represents every single page visit on the website. A page view referred from Google is given the same weight on the pie chart as a page view referred from another page on the site (e.g. a visitor clicks "Pricing" on the homepage and is therefore "referred" by www.storj.io).

Another way to phrase this, is that 90% direct traffic in the graph you shared does not necessarily mean we have a crazy/unreasonable amount of bot traffic. Rather, it means that for every one page visit that comes from an external source (e.g. Google, Twitter, whatever), there are nine page views due to internal navigation on the site.

We are already trying to remove bots from our metrics, but there could still be crawlers that come to our site that may explain some of this traffic. But the traffic could also be due to a situation where I came to www.storj.io from Google, then visited a bunch of pages on www.storj.io by clicking around.

TLDR:
"Top referral sources" implies that we mostly care about external traffic coming into the website, but the original query includes external _and_ internal traffic, which will result in a big chunk of direct traffic in the pie chart, even if we were able to completely remove bots from the metrics.

I have duplicated the query and made a version which excludes direct traffic. I believe this version will be a lot more useful for comparing external referral sources against each other.

https://metabase.datasci.storj.io/question/408-top-referral-sources-pie-chart-no-direct-traffic

Please let me know if this does not resolve your concerns.

**Comment by stoweandi** [comment]
Thanks @mobyvb! 

@prerna-parashar / @grafael / @Nastaja-Johnson Here is a _higher_ priority ticket for removing internal traffic from this query. @Nastaja-Johnson can you please trade out one of the other metric revision tickets for this one?
https://github.com/storj/datascience/issues/121
